### PR TITLE
Issue3438 d xdehumidifier

### PR DESCRIPTION
**Pull request by karthikeyad-pnnl**

### CODE REVIEW

**Comment by karthikeyad-pnnl** [comment]
@mwetter @JayHuLBL I am submitting this PR to merge the DX dehumidifier component model into the Buildings library. This is a fairly simple model with dependencies on classes already in the master branch. Please review the code as well as the proposed package structure for the new addition.
Thanks!
-Karthik

File: Buildings/Fluid/Humidifiers/DXDehumidifier.mo [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+  parameter Buildings.Fluid.Humidifiers.Data.Generic per
+    "Performance data"
+    annotation (choicesAllMatching=true,
+      Placement(transformation(extent={{60,-56},{80,-36}})));
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal
+    "Nominal mass flow rate";
+
+  parameter Modelica.Units.SI.PressureDifference dp_nominal
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/DXDehumidifier.mo:25]: add `displayUnit="Pa"` attribute.

File: Buildings/Fluid/Humidifiers/DXDehumidifier.mo [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    final min=0) = 3.412
+    "Rated energy factor, in L/kwh"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Buildings.Fluid.Humidifiers.Data.Generic per
+    "Performance data"
+    annotation (choicesAllMatching=true,
+      Placement(transformation(extent={{60,-56},{80,-36}})));
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/DXDehumidifier.mo:22]: group the parameter to group `Nominal condition`.

File: Buildings/Fluid/Humidifiers/DXDehumidifier.mo [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    "Transfer power and heat to the fluid medium";
+
+  parameter Real eneFac_nominal(
+    final min=0) = 3.412
+    "Rated energy factor, in L/kwh"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Buildings.Fluid.Humidifiers.Data.Generic per
+    "Performance data"
+    annotation (choicesAllMatching=true,
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/DXDehumidifier.mo:19]: make sure the parameters in the same group consequently together. 

For example here, the parameter `V_flow_nominal` is in group `Nominal condition`, the parameter `addPowerToMedium` is not grouped, and then again, the parameter `eneFac_nominal` is in group `Nominal condition` again. It should be avoided.

File: Buildings/Fluid/Humidifiers/DXDehumidifier.mo [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+  Modelica.Thermal.HeatTransfer.Sensors.HeatFlowSensor heaFloSen
+    "Sensor for measuring heat flow rate into medium"
+    annotation (Placement(transformation(extent={{-20,50},{0,70}})));
+
+  Modelica.Thermal.HeatTransfer.Sensors.TemperatureSensor senTem
+    "Temperature sensor"
+    annotation (Placement(transformation(extent={{38,50},{58,70}})));
+
+  Modelica.Blocks.Interfaces.RealOutput T(
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/DXDehumidifier.mo:45]: best should make sure the connectors are in the same sections of the code. The ideal order would be parameter section, connector sections, and other instances section.

**Review by JayHuLBL — CHANGES_REQUESTED** [review]
@karthikeyad-pnnl Please see inline comments.

Also, see following errors:
```
jianjunhu@ubuntu:~/GitFolder/karthikeyad-pnnl/modelica-buildings/Buildings$ ../bin/verifyFiles.py 
*** Error:  File 'Fluid/Humidifiers/Examples/DXDehumidifier.mo' contains invalid string 'Algorithm='.
*** Error:  File 'Fluid/Humidifiers/Examples/DXDehumidifier.mo' contains invalid string '__Dymola_Algorithm'.
*** Error:  File 'Fluid/Humidifiers/Validation/DXDehumidifier.mo' contains invalid string 'Algorithm='.
*** Error:  File 'Fluid/Humidifiers/Validation/DXDehumidifier.mo' contains invalid string '__Dymola_Algorithm'.
*** Terminating due to found errors in examined files.

jianjunhu@ubuntu:~/GitFolder/karthikeyad-pnnl/modelica-buildings/Buildings$ ../bin/validateHTML.py 
The following malformed html syntax has been found:
[-- ./Fluid/Humidifiers/DXDehumidifier.mo ]
line 69 column 1 - Warning: inserting implicit <p>
line 65 column 1 - Warning: trimming empty <p>
line 69 column 1 - Warning: trimming empty <p>
```

**Comment by karthikeyad-pnnl** [comment]
@JayHuLBL We have addressed the comments above. Please review it when you get a chance. Thanks!

File: Buildings/Fluid/Humidifiers/DXDehumidifier.mo [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+</html>",
+revisions="<html>
+<ul>
+<li>
+June 20, 2023, by Xing Lu:<br/>
+First implementation.
+</li>
+</ul>
+</html>"),
+    Icon(coordinateSystem(extent={{-100,-80},{100,80}}), graphics={
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/DXDehumidifier.mo:236]: change the `extent={{-100,-80},{100,80}}` to `extent={{-100,-100},{100,100}}`. So the Icon is consistent with other humidifiers like `Humidifier_u`.

File: Buildings/Fluid/Humidifiers/Data/Generic.mo [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+June 20, 2023, by Xing Lu:<br/>
+First implementation.
+</li>
+</ul>
+</html>"),
+    Icon(graphics={
+        Text(
+          extent={{-91,1},{-8,-54}},
+          textColor={0,0,255},
+          fontSize=16,
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/Data/Generic.mo:50]: remove the specification of the `fontSize`.

File: Buildings/Fluid/Humidifiers/Data/Generic.mo [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+          textString="watRem"),
+        Text(
+          extent={{2,-16},{94,-38}},
+          textColor={0,0,255},
+          textString="eneFac",
+          fontSize=16)}));
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/Data/Generic.mo:56]: remove the specification of the `fontSize`.

File: Buildings/Fluid/Humidifiers/Examples/Data/DXDehumidifier.mo [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+</html>",
+revisions="<html>
+<ul>
+<li>
+June 20, 2023, by Xing Lu and Karthik Devaprasad:<br/>
+First implementation.
+</li>
+</ul>
+</html>"),
+    Icon(graphics={
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/Examples/Data/DXDehumidifier.mo:24]: There is no need to have the `Icon()` annotation, as the class extents the `Buildings.Fluid.Humidifiers.Data.Generic`.

File: Buildings/Fluid/Humidifiers/Validation/DXDehumidifier.mo [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+model DXDehumidifier
+  "Validation model for DX dehumidifier"
+  extends Modelica.Icons.Example;
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal=0.13545
+    "Nominal mass flow rate";
+
+  parameter Modelica.Units.SI.Time tStepAve = 3600
+    "Time-step used to average out Modelica results for comparison with EPlus 
+    results.";
```
Comment (JayHuLBL) [inline Buildings/Fluid/Humidifiers/Validation/DXDehumidifier.mo:11]: Don't change to new line. Always keep the comment in one line.

**Review by JayHuLBL — CHANGES_REQUESTED** [review]
@karthikeyad-pnnl Please see inline comments.

**Comment by karthikeyad-pnnl** [comment]
@JayHuLBL: @terrancelu92 has addressed the pending comments. Please review the PR when you get a chance. Thanks!

**Pull request merged**